FWCore/Modules/src/GenericConsumer.cc

```diff
@@ -1,23 +1,108 @@
 /*
- *  This plugin depends on all the event, lumi and run products produced by the modules listed in its configuration:
- *    - eventProducts: depend on the event products from these modules
- *    - lumiProducts:  depend on the lumi products from these modules
- *    - runProducts:   depend on the run products from these modules
+ *  This EDAnalyzer will depend on all the event, lumi, run or process products declared by its configuration, both
+ *  transient and persistent.
  *
- *  Use "*" to depend on all the products in a given branch.
+ *  The dependencies can be specified either as module labels (e.g. "<module label>") or as branch names (e.g. 
+ *  "<product type>_<module label>_<instance name>_<process name>").
+ *  If a module label is used, no underscore ("_") must be present; this module will depend all the products produced
+ *  by that module, including automatic copy-to-host products in the case of Alpaka-based modules.
```

Comment:
I'd add the name of the more general feature behind the implicit copy-to-host ability
```suggestion
 *  by that module, including those produced by the Transformer ability (such as the implicitly copied-to-host products in case of Alpaka-based modules)
```
(same for the description in `fillDescriptions()`)

Reply:
Done, and added an example for the alpaka case.